src/charm_state.py

```diff
@@ -101,6 +104,22 @@ def _get_credentials_from_agent_relation(
     return Credentials(address=address, secret=secret)
 
 
+def get_agent_interface_dict_from_metadata(agent_meta: AgentMeta) -> dict:
+    """Generate dictionary representation of agent metadata.
+
+    Args:
+        agent_meta: The agent metadata object.
+
+    Returns:
+        A dictionary adhering to jenkins_agent_v0 interface.
+    """
+    return {
+        "executors": str(agent_meta.num_executors),
+        "labels": agent_meta.labels,
+        "name": agent_meta.name,
+    }
```

Comment:
Can't we use `model_dump` method from pydantic `BaseModel` ?
See https://docs.pydantic.dev/latest/api/base_model/#pydantic.main.BaseModel.model_dump

Reply:
That's a good point, thanks, I was thinking about how to cleanly build the relation interface dict. I added a serializer and used `model_dump()` in the charm and observer code.

Reply:
in the end pydantic v2 is too problematic to be used for now due to the fact that `pydantic_core` wheel needs to be built with rust and as a result, requires the charm to be built on their respective series to avoid issues (charm that runs on 22.04 need to be built on 22.04 with python 3.10, same for 20.04) -> This causes our CI to complain since we are generating 2 `.charm` files.

With that much overhead I'm reverting back to using a method to convert this to a dict